### ISSUE TITLE
Cant jit the function using `soft_sort.quantile()` and `jnp.where()`

### ISSUE THREAD

**Issue by michalk8** [ott-jax/ott#407]
### Discussed in https://github.com/ott-jax/ott/discussions/406

<div type='discussions-op-text'>

<sup>Originally posted by **yyang97** August  1, 2023</sup>
```
import jax
import ott
import jax
import jax.random
import jax.numpy as jnp

N = 200
x = jax.random.normal(key = jax.random.PRNGKey(0), shape=(N,))
a = 0.5
quant = jnp.array([.2,.8])
def quantile_cal(x,quant,a):
   return( jax.vmap(lambda quant: ott.tools.soft_sort.quantile(jnp.where(x < a, x, a), 
                                                                    q=quant))(quant))


quantile_cal(x,quant,a)

# Array([[-0.81860286],
#        [ 0.49906105]], dtype=float32)
```

If I don't jit the function, it looks fine. 

But if I jit it, it fails:

```
@jax.jit
def quantile_cal(x,quant,a):
   return( jax.vmap(lambda quant: ott.tools.soft_sort.quantile(jnp.where(x < a, x, a), 
                                                                    q=quant))(quant))


quantile_cal(x,quant,a)


# NonConcreteBooleanIndexError: Array boolean indices must be concrete; got ShapedArray(bool[3])

# See https://jax.readthedocs.io/en/latest/errors.html#jax.errors.NonConcreteBooleanIndexError
```

However, I don't have any array boolean things...

Further, it seems that ott package about the quantile function is updated... The same code works before...</div>

**Comment by michalk8** [comment]
Hi @yyang97, will be fixed in #408, just tests need to pass; after you should install `ott-jax` from `main`.
Furthermore, after #382 introduced by @marcocuturi , you don't need to (and shouldn't) vmap over the quantiles; the below code will just work:

```python
import jax
import jax.numpy as jnp
from ott.tools import soft_sort

N = 2000
x = jax.random.normal(key = jax.random.PRNGKey(0), shape=(N,))
a = 0.5
quant = jnp.array([.2,.8])

@jax.jit
def quantile_cal(x, quant, a):
   return soft_sort.quantile(jnp.where(x < a, x, a), q=quant)

quantile_cal(x, quant, a)
```

**Comment by yyang97** [comment]
Thank you very much!!! This function is just very important to me :)

**Comment by michalk8** [comment]
I also noticed that the vmapped version is computing wrong values, at the moment, am not sure why... just running:
```
import numpy as np
np.quantile(np.where(x < a, x, a), q=quant)

# array([-0.87661878,  0.5       ])
```
whereas the vmapped function returned
```python
# Array([[-0.81860286],
#        [ 0.49906105]], dtype=float32)
```

When passing `q=jnp.array([.2,.8])`, I get `Array([-0.87872696,  0.4870388 ], dtype=float32)`, which seems correct to the actual "hard" quantiles. Will investigate in another issue.
